### PR TITLE
#530555: [SXA][Headless] Redirects with defined language in source URL don't work

### PR DESCRIPTION
**Pull request by AntonKechashin**
SXA Redirects middleware now checks the patterns for registered locales and updates the URL parameters accordingly.

### CODE REVIEW

File: packages/create-sitecore-jss/src/templates/nextjs-sxa/src/lib/middleware/plugins/redirects.ts [hunk 1]
```diff
@@ -13,6 +15,7 @@ class RedirectsPlugin implements MiddlewarePlugin {
       apiKey: config.sitecoreApiKey,
       siteName: config.jssAppName,
     });
+    this.locales = nextConfig().i18n.locales;
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs-sxa/src/lib/middleware/plugins/redirects.ts:18]: `locales` should be owned by `RedirectsMiddleware` (another option on `RedirectsMiddlewareConfig`)

**Review by ambrauer — CHANGES_REQUESTED** [review]
See a couple suggestions

**Review by ambrauer — CHANGES_REQUESTED** [review]
See one comment and also please check lint errors on `sitecore-jss-nextjs` package.

File: packages/sitecore-jss-nextjs/src/edge/redirects-middleware.ts [hunk 2]
```diff
@@ -20,14 +20,16 @@ export type RedirectsMiddlewareConfig = Omit<GraphQLRedirectsServiceConfig, 'fet
...
   private redirectsService: GraphQLRedirectsService;
+  private locales: string[];
 
   /**
    * NOTE: we provide native fetch for compatibility on Next.js Edge Runtime
    * (underlying default 'cross-fetch' is not currently compatible: https://github.com/lquixada/cross-fetch/issues/78)
    * @param {RedirectsMiddlewareConfig} [config] redirects middleware config
    */
-  constructor(config: RedirectsMiddlewareConfig) {
+  constructor(config: RedirectsMiddlewareConfig, locales: string[]) {
```
[the remark this answers is not in the document]
Comment (AntonKechashin) [reply packages/sitecore-jss-nextjs/src/edge/redirects-middleware.ts]: Moved `locales` to `RedirectsMiddlewareConfig`

**Pull request merged**